modules/rabbitmq/src/main/java/org/testcontainers/containers/RabbitMQContainer.java

```diff
@@ -0,0 +1,358 @@
+        return String.format("http://" + getContainerIpAddress() + ":" + getMappedPort(DEFAULT_HTTP_PORT));
+    }
+
+    /**
+     * @return URL of the HTTPS management endpoint.
+     */
+    public String getHttpsUrl() {
+        return String.format("https://" + getContainerIpAddress() + ":" + getMappedPort(DEFAULT_HTTPS_PORT));
+    }
+
+    /**
+     * Sets the password for the admin (default is <pre>guest</pre>)
+     *
+     * @param adminPassword The admin password.
+     * @return This container.
+     */
+    public S withAdminPassword(final String adminPassword) {
+        this.adminPassword = adminPassword;
+        return self();
+    }
+
+    public enum SslVerification {
+        VERIFY_NONE("verify_none"), VERIFY_PEER("");
+
+        SslVerification(String value) {
+            this.value = value;
+        }
+
+        private final String value;
+    }
+
+    public S withSSL(
+            final MountableFile keyFile,
+            final MountableFile certFile,
+            final MountableFile caFile,
+            SslVerification verify,
+            boolean failIfNoCert,
+            int verificationDepth) {
+
+        return withSSL(keyFile, certFile, caFile, verify, failIfNoCert)
+                .withEnv("RABBITMQ_SSL_DEPTH", String.valueOf(verificationDepth));
+    }
+
+    public S withSSL(
+            final MountableFile keyFile,
+            final MountableFile certFile,
+            final MountableFile caFile,
+            SslVerification verify,
+            boolean failIfNoCert) {
+
+        return withSSL(keyFile, certFile, caFile, verify)
+                .withEnv("RABBITMQ_SSL_FAIL_IF_NO_PEER_CERT", String.valueOf(failIfNoCert));
+    }
+
+    public S withSSL(
+            final MountableFile keyFile,
+            final MountableFile certFile,
+            final MountableFile caFile,
+            SslVerification verify) {
+
+        return withEnv("RABBITMQ_SSL_CACERTFILE", "/etc/rabbitmq/ca_cert.pem")
+                .withEnv("RABBITMQ_SSL_CERTFILE", "/etc/rabbitmq/rabbitmq_cert.pem")
+                .withEnv("RABBITMQ_SSL_KEYFILE", "/etc/rabbitmq/rabbitmq_key.pem")
+                .withEnv("RABBITMQ_SSL_VERIFY", verify.value)
+                .withCopyFileToContainer(certFile, "/etc/rabbitmq/rabbitmq_cert.pem")
+                .withCopyFileToContainer(caFile, "/etc/rabbitmq/ca_cert.pem")
+                .withCopyFileToContainer(keyFile, "/etc/rabbitmq/rabbitmq_key.pem");
+    }
+
+    public S withPluginsEnabled(String... pluginNames) {
+        List<String> command = new ArrayList<>(asList("rabbitmq-plugins", "enable"));
+        command.addAll(asList(pluginNames));
+        command.add("--offline");
+        return self();
+    }
+
+    public S withPluginsDisabled(String... pluginNames) {
+        List<String> command = new ArrayList<>(asList("rabbitmq-plugins", "disable"));
+        command.addAll(asList(pluginNames));
+        command.add("--offline");
+        values.add(command);
```

Comment:
`withPluginsDisabled("foo").withPluginsDisabled("bar")` will break this code.
I would suggest storing the disabled/enabled plugins and form the command in `configure`

Reply:
Fixed by removing withPluginsDisabled(). Not sure why you would need it anyways.

Reply:
I don't think this is resolved, unless RabbitMQ Admin gracefully handles `withPluginsEnabled("foo").withPluginsEnabled("foo")`

Reply:
I handles that gracefully. This works fine:
```
new RabbitMQContainer()
.withPluginsEnabled("rabbitmq_shovel")
.withPluginsEnabled("rabbitmq_shovel");

assertThat(container.execInContainer("rabbitmq-plugins", "is_enabled", "rabbitmq_shovel", "--quiet").getStdout()).contains("rabbitmq_shovel is enabled");
```

It will execute _rabbitmq-plugins enable rabbitmq_shovel_ twice, but I don't think that's a deal breaker.